### ISSUE TITLE
Using JS Interop in OnInitializedAsync() cannot initially find JS function #29215

### ISSUE THREAD

**Issue by ShingYehGit** [dotnet/aspnetcore#58174]
The Following Blazor/Wasm .Net.8 VS2022 will work as exppected for JavaScript InterOp

1. in Index.html as follow
```
<body>
    <script src="_framework/blazor.webassembly.js"></script>
     <!-- The following code work will -->
    <script src="SiteScript.js"></script>
    <!-- Won't won't work -->
    <!-- script src="SamuelScript/WangScript.js" --></script>
</body>
```

2. In SiteScript.js
```
function helloFunction() {
    alert("Hello Blazor Web Assembly");
}

function showAlert(name) {
    alert('Hello ' + name);
}
```
4. In Page razor file OnInitializedAsync() work well as ecpected
    ```
    @inject IJSRuntime JSRuntime

    <button @onclick="CallJavaScript">Click Me</button>

    protected override async Task OnInitializedAsync()
    {
        var jsInProcess = (IJSInProcessRuntime)JS;
        await jsInProcess.InvokeVoidAsync("showAlert", "Samuel Wang");//Work
    }

    private async Task CallJavaScript()
    {
        //WORK await JS.InvokeVoidAsync("helloFunction");
        // NOT WORK await JS.InvokeVoidAsync("showAler", "Samuel Wang");

        var jsInProcess = (IJSInProcessRuntime)JS;
        //jsInProcess.InvokeVoid("showAlert", "Unkown user");//Work
        await jsInProcess.InvokeVoidAsync("showAlert", "Unkown user");//Work
        //jsInProcess.InvokeVoidAsync("helloFunction");//WORK
    }
    ```

**Comment by MackinnonBuck** [comment]
Thanks for reaching out, @ShingYehGit.

It looks like the line marked as "NOT WORK" has a typo (`showAler`, which should be `showAlert`). If you fix that typo, does the code work as expected?

**Comment by mkArtakMSFT** [comment]
Repro app from the customer: https://github.com/ShingYehGit/WasmSkiaApp/tree/master

**Comment by mkArtakMSFT** [comment]
Looks like you're missing a closing curly bracket `}` in the SiteScript.js file (the last function isn't closed). And that's causing the function not to be found later. This is for the closed / duped issue that you've filed: #58071

**Comment by ShingYehGit** [comment]
Many Thanks for the tip!

Artak ***@***.***> 於 2024年10月4日 週五 下午12:54寫道：
